src/fsharp/NameResolution.fs

```diff
@@ -2483,11 +2499,81 @@ let ResolveFieldPrim (ncenv:NameResolver) nenv ad typ (mp,id:Ident) =
         if isAppTy g typ then 
             match ncenv.InfoReader.TryFindRecdOrClassFieldInfoOfType(id.idText,m,typ) with
             | Some (RecdFieldInfo(_,rfref)) -> [ResolutionInfo.Empty, FieldResolution(rfref,false)]
-            | None -> error(Error(FSComp.SR.nrTypeDoesNotContainSuchField((NicePrint.minimalStringOfType nenv.eDisplayEnv typ), id.idText),m))
+            | None ->
+                // record label doesn't belong to record type
```

Comment:
Again, move all this to a separate top-level function with a systematic name like ``SuggestFieldPrim`` or ``ResolveFieldPrimFailure`` or the like.